src/craftdef.cpp

```diff
@@ -960,6 +960,81 @@ class CCraftDefManager: public IWritableCraftDefManager
 
 		return recipes;
 	}
+
+	virtual bool clearCraftRecipesByOutput(CraftOutput &output, IGameDef *gamedef)
+	{
+		std::map<std::string, std::vector<CraftDefinition*> >::iterator vec_iter = m_output_craft_definitions.find(output.item);
+
+		if (vec_iter == m_output_craft_definitions.end())
+			return false;
+
+		std::vector<CraftDefinition*> &vec = vec_iter->second;
+		for (std::vector<CraftDefinition*>::iterator i = vec.begin();
+			i != vec.end(); ++i) {
+			CraftDefinition *def = *i;
+			std::vector<CraftDefinition*> &unhashed_inputs_vec = m_craft_defs[(int) CRAFT_HASH_TYPE_UNHASHED][0]; // Recipes are not yet hashed at this point
+			std::vector<CraftDefinition*> new_vec_by_input;
+			for (std::vector<CraftDefinition*>::iterator i2 = unhashed_inputs_vec.begin();
+				i2 != unhashed_inputs_vec.end(); ++i2) {
+				if(def != *i2)
+				{
+					new_vec_by_input.push_back(*i2);
+				}
+			}
+			m_craft_defs[(int) CRAFT_HASH_TYPE_UNHASHED][0].swap(new_vec_by_input);
+		}
+		m_output_craft_definitions.erase(output.item);
+		return true;
+	}
+
+	virtual bool clearCraftRecipesByInput(CraftMethod craft_method, unsigned int craft_grid_width, const std::vector<std::string> &recipe, IGameDef *gamedef)
+	{
+		bool all_empty = true;
+		for (std::vector<std::string>::size_type i = 0;
+			i < recipe.size(); i++) {
```

Comment:
Two tabs indent, not one (to keep it on a separate ident level than the content below)